### PR TITLE
Mega Menu

### LINKED ISSUE #177: Logotype text can't be clicked
In the main layout, you have the logotype with the 'F' and 'teknologsektionen inom TLTH'. You can click the F but not the text. Probably because some floating element on the right is overlapping it.

It's a small thing but it really grinds my gears...

### PR DESCRIPTION
**Pull request by navro**
* Removes the hard coded main menu
* Adds main_menu model/controller
* Adds two new types of menus: Mega Menu and Mega Menu (full width)
* Fixes footer width overflow
* Fixes menu width overflow
* Fixes footer position
* Fixes #177

### CODE REVIEW

File: app/assets/stylesheets/partials/_menus.scss [hunk 1]
```diff
@@ -1,9 +1,69 @@
-.dropdown-menu {
-  .menu {
-    i {
-      float: right;
-      margin-top: 4px;
+.menu-large {
+  position: static !important;
```
Comment (davidwessman) [inline app/assets/stylesheets/partials/_menus.scss]: Hmm, it is better if you are more descriptive, perhaps use more than one class:
```scss
.dropdown {
  &.menu-large {
    position: static;
  }
}
```

File: app/assets/stylesheets/partials/_menus.scss [hunk 2]
```diff
@@ -1,9 +1,69 @@
...
+}
+
+.megamenu {
+  padding: 10px 0;
+
+  &.fw {
```
Comment (davidwessman) [inline app/assets/stylesheets/partials/_menus.scss]: I think `full-width`is more descriptive, but I guess this is only used in one partial view.

File: app/views/layouts/_menu.html.erb [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
   <ul class="nav navbar-nav navbar-right">
-    <% cache_unless(can_administrate?(:manage, Menu), ['main_menu', I18n.locale]) do %>
-      <% global_menus = GlobalMenus.new %>
-      <%= menu_dropdown(menus: global_menus.guild_menus, title: t('model.menus.guild')) %>
-      <%= menu_dropdown(menus: global_menus.member_menus, title: t('model.menus.members')) %>
-      <%= menu_dropdown(menus: global_menus.company_menus, title: t('model.menus.companies')) %>
-      <%= menu_dropdown(menus: global_menus.contact_menus, title: t('model.menus.contact')) %>
+    <% cache(['main_menu', I18n.locale], skip_digest: true) do %>
+      <% MainMenu.includes(:translations).index.each do |m| %>
+        <% if m.mega %>
```
Comment (davidwessman) [inline app/views/layouts/_menu.html.erb]: Put row 5-20 in a partial called `_mega_menu` and row 22-27 in a partial called `_menu` and then create a method for the partial path in `MainMenu` and then you can just say
```ruby
render(MainMenu.includes(:translations).index)
```

File: app/view_objects/global_menus.rb [hunk 4]
```diff
@@ -1,11 +1,3 @@
 class GlobalMenus
```
Comment (davidwessman) [inline app/view_objects/global_menus.rb]: Remove this one completely :)

**Comment by davidwessman** [comment]
Great work!
Just some small comments, otherwise :+1:

File: app/views/layouts/_menu.html.erb [hunk 5]
```diff
@@ -1,11 +1,32 @@
...
+          <li class="dropdown menu-large">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              <%= safe_join([m, ' ', fa_icon('angle-down')]) %>
+            </a>
+            <ul class="dropdown-menu megamenu row <%= m.fw ? 'fw' : '' %>">
+              <% menu = m.menus.includes(:translations).index %>
+              <% col = menu.select(:column).distinct(:column).count %>
```
Comment (davidwessman) [inline app/views/layouts/_menu.html.erb]: The whole `col` logic should perhaps be in a helper.
Is the amount of columns limited? 1-4?

File: app/assets/javascripts/footer.js [hunk 6]
```diff
@@ -8,7 +8,7 @@ bottom = function () {
     var footerPadding = parseInt($footer.css('padding-top')) +
                         parseInt($footer.css('padding-bottom'));
 
-    var diff = docHeight - copyrightBottom + footerPadding;
+    var diff = docHeight - copyrightBottom + footerPadding - 28;
```
Comment (davidwessman) [inline app/assets/javascripts/footer.js]: This wasn't working anymore? :/

Comment (navro) [inline app/assets/javascripts/footer.js]: > This wasn't working anymore? :/

No, it has not worked for a long time. I'm not sure where 28 actually comes from, but it fixes the problem

File: app/views/admin/menus/_form.html.erb [hunk 7]
```diff
@@ -1,16 +1,15 @@
 <%= simple_form_for([:admin, menu]) do |f| %>
-  <%= f.input :location, collection: [[t('model.menus.guild'), Menu::GUILD],
-                                      [t('model.menus.members'), Menu::MEMBER],
-                                      [t('model.menus.companies'), Menu::COMPANY],
-                                      [t('model.menus.contact'), Menu::CONTACT]],
-                         input_html: { class: 'select2' },
-                         include_blank: false %>
+  <%= f.input :main_menu_id, collection: MainMenu.all,
```
Comment (davidwessman) [inline app/views/admin/menus/_form.html.erb]:
```ruby
f.association :main_menu, collection:...
```
(Unless there is some other magic that I do not see here at least 😄)

File: app/views/layouts/_copyright.html.erb [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
   <div class="container">
     <div class="row">
-      <div class="col-md-6 col-sm-12">
+      <div class="col-xm-12 col-sm-6">
         <p class="year">
           <%= Time.current.year %> &copy; <%= t('.the_f_guild') %>.
         </p>
       </div>
-      <div class="col-md-6 col-sm-12">
+      <div class="col-xm-12 col-sm-6">
```
Comment (davidwessman) [inline app/views/layouts/_copyright.html.erb]: What is `col-xm-12`?

File: app/views/layouts/_menu.html.erb [hunk 9]
```diff
@@ -1,11 +1,9 @@
...
   <ul class="nav navbar-nav navbar-right">
-    <% cache_unless(can_administrate?(:manage, Menu), ['main_menu', I18n.locale]) do %>
-      <% global_menus = GlobalMenus.new %>
-      <%= menu_dropdown(menus: global_menus.guild_menus, title: t('model.menus.guild')) %>
-      <%= menu_dropdown(menus: global_menus.member_menus, title: t('model.menus.members')) %>
-      <%= menu_dropdown(menus: global_menus.company_menus, title: t('model.menus.companies')) %>
-      <%= menu_dropdown(menus: global_menus.contact_menus, title: t('model.menus.contact')) %>
+    <% cache(['main_menu', I18n.locale], skip_digest: true) do %>
+      <% MainMenu.includes(:translations).index.each do |m| %>
+        <%= render(m.partial_path, main_menu: m) %>
```
Comment (davidwessman) [inline app/views/layouts/_menu.html.erb]: Nice!

File: app/views/layouts/_copyright.html.erb (hunk 8 above)
Comment (navro) [inline app/views/layouts/_copyright.html.erb]: > What is col-xm-12?

Like col-sm-12, but for even smaller screens. sm is tablets, xm is phones

Comment (davidwessman) [inline app/views/layouts/_copyright.html.erb]: Hm, shouldn't it be `xs`?
http://getbootstrap.com/css/#grid-options

Comment (navro) [inline app/views/layouts/_copyright.html.erb]: > Hm, shouldn't it be xs?
> http://getbootstrap.com/css/#grid-options

Typo 😉

**Comment by davidwessman** [comment]
Merge merge 👯

**Pull request merged**